pkg/operator/operator.go

```diff
@@ -90,6 +92,9 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 	azConfig, err := GetAZConfig()
 	lo.Must0(err, "creating Azure config") // NOTE: we prefer this over the cleaner azConfig := lo.Must(GetAzConfig()), as when initializing the client there are helpful error messages in initializing clients and the azure config
 
+	// Get a token to ensure we can
+	lo.Must0(ensureToken(azConfig), "ensuring Azure token can be retrieved")
```

Comment:
I was torn if we should actually use the token here, or just get one. I went with just get one as it seemed slightly more targeted and also easier to truly assert than "the token is usable" -- because the token being usable gets at permissions and so would realistically require more than a single check to assert all of the permissions that we need.

With that said, the downside here is that we can retrieve the token but can't use it at all... which isn't _quite_ what we were doing before with the vnetGUID stuff, where we actually used the token to do something.

Reply:
So I could use the token to list LBs at the NRG level for example and just throw the response away

Reply:
Just getting one seems good enough / a good compromise to me, given the kind of failures we are fixing here

Reply:
getting the token once is fine as there are multiple cache layers in IMDS / managed environment. It would be great if we could reuse the azidentity instance created in this ensure call as azidentity has its own cache layer too.